Bug/issue 3370 - Unable to opt out of MetaMetrics (backup)

Fatxx opened the pull request:
<!--
Thanks for your contribution!

Please ensure that any applicable requirements below are satisfied before submitting this pull request. This will help ensure a quick and efficient review cycle.
-->

**Description**

Fixing issue where Metametrics option in settings was not persisted between application sessions. 

**Checklist**

* [X] There is a related GitHub issue
* [ ] Tests are included if applicable
* [X] Any added code is fully documented

**Issue**

Resolves #3370

Fatxx committed:
Persist anaylitcs enabled state in redux

Fatxx committed:
Updated SecuritySettings snapshots

Fatxx committed:
Remove unecessary comment

Fatxx commented:
I have read the CLA Document and I hereby sign the CLA

Fatxx commented:
@sethkfman basically, I moved the flag to the redux store and removed the `analytics` store from the blacklist to persist the value.

sethkfman commented:
@Fatxx We had added analytics to the redux persist blacklist because we are trying to move towards using more native stores for persisting simple data. I think you solution would work great but we want to move away from redux for these simple persisted data types. 

We are using DefaultPreferences to persist this value and must have missed this update. This is were the metrics are first set when the user onboards: https://github.com/MetaMask/metamask-mobile/blob/1647241bd25d7b5e38400bfa10a66e01f41c6dd8/app/components/UI/OptinMetrics/index.js#L207

I think we just need to initialize this value in the view and make sure the Analytics class also uses it.

Fatxx commented:
@sethkfman  A question that arises on my side is that you're alerting the user that a restart is required. However, as far as I'm concerned, it seems that the tracking is disabled right after `Analytics.disable()` is executed since the flag inside the `Analytics` instance changes to `false,` meaning that the `_trackEvent` never fires an `RCTAnalytics.trackEvent` anymore.

Is there any hidden reason for this that I'm not aware of?

Commit:
Remove analytics store and migrate data to UserPreference

Fatxx committed:
WIP

Fatxx committed:
WIP

Fatxx committed:
Remove redundant AnalyticsService

Fatxx committed:
Remove redundant AnalyticsService

Fatxx commented:
@sethkfman just pushed new changes using `DefaultPreference` instead of Redux. 

To sum up, I basically removed `Analytics` from Redux since we now rely only on DefaultPreference as the source of truth, this preference is loaded on Analytics initialization which caches the flag value in memory in the class instance. I also removed the AnalyticsService since in my POV is redundant because we can simply init the Analytics directly.

I removed the user alert for now since it's a false positive of what's actually happening.

sethkfman committed:
Merge branch 'develop' into bug/issue-3370

sethkfman committed:
Merge branch 'develop' into bug/issue-3370

sethkfman commented on app/components/Views/Settings/SecuritySettings/index.js:
```diff
@@ -460,16 +459,12 @@ class Settings extends PureComponent {
 			await this.trackOptInEvent('Metrics Opt Out');
 			Analytics.disable();
-			this.setState({ metricsOptIn: false });
-			Alert.alert(
```
@Fatxx I spoke with Andre and we are pretty sure we need this Alert due some native initialization. I would keep the Alert for now.

Fatxx replied:
Ok sure, I'll add that.

sethkfman reviewed (commented):
Minor comment on the Alert

Fatxx committed:
Added alert back, fixed typo

Fatxx committed:
Added alert back, fixed typo

Fatxx committed:
Merge branch 'develop' into bug/issue-3370

Fatxx committed:
Merge branch 'develop' into bug/issue-3370

Fatxx committed:
:Merge branch 'bug/issue-3370' of github.com:Fatxx/metamask-mobile into bug/issue-3370

Fatxx committed:
Merge branch 'MetaMask:develop' into bug/issue-3370
